Bike controller overhaul

yapoov opened the pull request:
- Player controller now uses two interfaces to get controller input (keyboard, MQTT, joystick etc.) to move the bike according to the bike movement handle. Right now it has SimpleBikeController and RealisticBikeController.

- Added Avatar ik controller to place the avatars hands on the bike handle, and feet on the pedals.

- The avatar model had incorrect rig hierarchy, fixed the hierarchy, now that any humanoid animation can work on the player avatar model.

- Added bike movement handler type select menu in the garage scene settings menu

- Set the npc model to use the same system as the player, this was trivial as the npc only needs to provide the input and use the existing bike movement handlers.

yapoov committed:
## Bike controller overhaul

- Player controller now uses two interfaces to get controller input (keyboard, MQTT, joystick etc.) to move the bike according to the bike movement handle. Right now it has SimpleBikeController and RealisticBikeController.

- Added Avatar ik controller to place the avatars hands on the bike handle, and feet on the pedals.

- The avatar model had incorrect rig hierarchy, fixed the hierarchy, now that any humanoid animation can work on the player avatar model.

- Added bike movement handler type select menu in the garage scene settings menu

- Set the npc model to use the same system as the player, this was trivial as the npc only needs to provide the input and use the existing bike movement handlers.

djasmit reviewed (changes requested):
A couple of notes;

1. This is missing the new_player prefabs meta files. This causes all links to that prefab to be broken, stopping scenes using the Network Manager from loading as they require those links. These files must be generated before accepting the PR.

2. There are modifications to the player prefab that are likely unnecessary as this is already using a new_player prefab. It would be best to omit those changes if possible.

3. This PR is quite large and changes a few not entirely related aspects. There could be some potential clashes, particularly with the NPC changes.

yapoov committed:
forgot to add player_new.meta

yapoov committed:
Merge remote-tracking branch 'upstream/main' into BikeMotionImprovement

# Conflicts:
#	Assets/Prefabs/Player.prefab
#	Assets/Scripts/Gameplay/PlayerController.cs

yapoov committed:
Fixed merge conflicts with playercontroller,
reverted Player.prefab changes.
Moved the GroundLockMode logic to the SimpleBikeController.cs script.

Thumbness reviewed (dismissed):
LGTM

yapoov committed:
Setting players rigidbody to have interpolation makes it spawn at vector3(0,0,0)

djasmit reviewed (changes requested):
Still need to resolve this conflict with the main player Prefab (just override it and use Orgil's new prefab in scenes). 

![image](https://github.com/user-attachments/assets/bfecd41e-2cf9-4bac-94af-6083fbd3ea01)

yapoov committed:
Merge branch 'main' into BikeMotionImprovement

djasmit reviewed (changes requested):
### **DO NOT ACCEPT THIS PR**

The merge has been successful, however this PR has created an entirely unforeseen issue that stops all missions from loading. May have to push this update to next trimester.

It will destroy the project until this issue is resolved.

![image](https://github.com/user-attachments/assets/010ed6a3-2daf-4acc-b04a-22f85c1f9e57)

yapoov committed:
fixed broken player reference in mission 2.

djasmit reviewed (approved):
I'm going to tentatively approve this, under the knowledge that this WILL break local copies of the Unity project and require EVERY owner to manually re-bake prefabs under the fusion tab. We'll reference this in handover.

yapoov committed:
Merge remote-tracking branch 'upstream/main' into BikeMotionImprovement

# Conflicts:
#	Assets/Scenes/CityScene.unity

yapoov committed:
-Added photon physics package to fix the movement stutternig issue
- Fixed merge conflict

yapoov committed:
Merge branch 'main' into BikeMotionImprovement

yapoov committed:
Set the photon physics client simulation mode to always

yapoov committed:
Merge remote-tracking branch 'origin/BikeMotionImprovement' into BikeMotionImprovement

yapoov committed:
Merge branch 'main' into BikeMotionImprovement
